tests/features/steps/artifacts-poetry-plugin-steps.py

```diff
@@ -0,0 +1,67 @@
+from behave import given
+from behave import when
+from behave import then
+from artifacts_poetry_plugin.plugin import getPackages
+from poetry.factory import Factory
+import os
+import shutil
+from pathlib import Path
+import subprocess
+
+TEST_RESOURCES_DIR = os.path.join(os.getcwd(), "tests", "resources")
+TEMPLATE_PY_PROJECT = os.path.join(TEST_RESOURCES_DIR, "template.toml")
+PY_PROJECT = os.path.join(TEST_RESOURCES_DIR, "pyproject.toml")
+POETRY_LOCK = os.path.join(TEST_RESOURCES_DIR, "poetry.lock")
+
+
+class Package:
+    def __init__(self, name, pretty_version) -> None:
+        self.name = name
+        self.pretty_version = pretty_version
+
+    def name(self):
+        return self.name
+
+    def pretty_version(self):
+        return self.pretty_version
+
+
+@given("the following artifacts")
+def the_following_artifacts(context):
+    context.packages = {}
+    for row in context.table:
+        context.packages[row["name"]] = Package(
+            name=row["package"], pretty_version=row["version"]
+        )
+
+
+@given("a Python project with dependencies on artifacts {keys}")
+def step_impl(context, keys):
```

Comment:
Doesn't seem that keys are used in this method.  I can see that the packages are being used for A & B and they are in the context.packages dict, but might want to be more explicit in loading packages A and B.  i.e. 

for key in keys:
    package = context.packages[key]

Reply:
Fixed.